### PR TITLE
Allow `xlink:href` to get removed, fix `option.value` in mock

### PR DESCRIPTION
**Pull request by isiahmeadows**
## Motivation and Context

Just noticed while playing around with some optimization that `xlink:*` attributes were being mishandled, but I also noticed that `option.value` was incorrect in the mock, so I fixed it.

## How Has This Been Tested?

Added some new tests and ran them all as usual.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have updated `docs/change-log.md`

### CODE REVIEW

File: render/tests/test-attributes.js [hunk 1]
```diff
@@ -376,15 +376,15 @@ o.spec("attributes", function() {
 
 			o(a.dom.value).equals("1")
 		})
-		o("null becomes the empty string", function() {
+		o("null becomes 'null'", function() {
```
Comment (tivac) [inline render/tests/test-attributes.js]: Why is this behavior desirable? I can't think of a time where you'd want to set `"null"` as an attribute value.

File: test-utils/tests/test-domMock.js [hunk 2]
```diff
@@ -954,11 +954,11 @@ o.spec("domMock", function() {
 					o(select.selectedIndex).equals(1)
 				}
 			})
-			o("option.value = null is converted to the empty string", function() {
+			o("option.value = null is converted to 'null'", function() {
```
Comment (tivac) [inline test-utils/tests/test-domMock.js]: See my previous question.

File: render/tests/test-attributes.js (hunk 1 above)
Comment (isiahmeadows) [reply render/tests/test-attributes.js]: Just matching our behavior to the [HTML](https://html.spec.whatwg.org/multipage/form-elements.html#dom-option-value) and [WebIDL](https://heycam.github.io/webidl/#es-DOMString) specs. (Notably, the DOM's `option.value` does *not* have the `[TreatNullAs=EmptyString]` extended attribute.)

Comment (isiahmeadows) [reply render/tests/test-attributes.js]: I filed #1978 (but closed it for reasons not related to this).

Comment (isiahmeadows) [reply render/tests/test-attributes.js]: If you feel strongly enough about it, I can fix it pretty quickly.

(FYI, the mock still needed fixed, though.)

Comment (pygy) [reply render/tests/test-attributes.js]: Actually this is in conflict with #1865 which I'm currently trying to rebase. I had set up on treating `null` and `undefined` in vdom indiscriminately as a lack of attribute, see https://github.com/MithrilJS/mithril.js/pull/1865/commits/fa53a66ab775c9661c594bb24b9baf6fd1528974#diff-0c7e3694f9865906ffe0847957a6c3a9

I'll bring it up in gitter as well for visibility.

**Pull request merged**